### ISSUE TITLE
Template_type missing issue 

### ISSUE THREAD

**Issue by alvin-reyes** [BotMill/fb-botmill#28]
It seems that all Event is having issues passing the template type.

```
02:29:15.738 ERROR [main][NetworkUtils] HTTP connection failed with error code 400.
02:29:15.739 ERROR [main][NetworkUtils] Error message from Facebook. Message: [(#100) Missing template_type for attachment's payload], Code: [100], Type: [OAuthException], FbTraceID: [DtlNZ2Ig9GY].
Sent!
```

Tested this with:
- List Template
- Generic Template
- Button Template

Please take a look.

**Comment by alvin-reyes** [comment]
Fixed now. It seems that the issue was my test. I got two message event with the same text message. The chain fails entirely because of this.

**Comment by aurasphere** [comment]
Nope, this was a real issue. The problem was the custom deserializer that I've introduced yesterday that somehow messed up with the serialization as well. It would only affect the outcoming messages with a payload (templates but not the text-only messages). I've fixed it with cc798fee3b3c4ead1e48a0bc8ce44eaa5207bb77 along with some minor bugs on the buttons.
